### PR TITLE
WIP: client,server: Expose reuse-port and reuse-address so they can be configured via a config file

### PR DESCRIPTION
**Pull request by lilic**
To be able to set socket options reuse-port and reuse-address to true in a configuration file, we need to expose these options first. 

@hexfusion while debugging something I noticed these can't be set via a configuration file, I exposed it as I needed it for some testing, but let me know if this is not how you envisioned it to be exposed ever? Thank you!

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1,35 +1,35 @@
 #!/usr/bin/env bash
 #
 # Run all etcd tests
-# ./test
-# ./test -v
+# ./test.sh
```
Comment (lilic) [inline test.sh:4]: Let me know if you want me to do in another PR? I just noticed when trying to run some unit tests.

Comment (ptabor) [reply test.sh:4]: Feel free to smuggle it ;)

File: server/embed/config_test.go [hunk 2]
```diff
@@ -72,6 +76,14 @@ func TestConfigFileOtherFields(t *testing.T) {
 		t.Errorf("ForceNewCluster = %v, want %v", cfg.ForceNewCluster, true)
 	}
 
+	if cfg.SocketOpts.ReusePort == false {
```
Comment (ptabor) [inline server/embed/config_test.go:79]: Optional: 
I found 
```
assert.Equals(t, true, cfg.SocketOpts.ReusePort, "ReusePort does not match") 
```
shorter and providing a nice output in case of failure.

File: server/embed/config_test.go [hunk 3]
```diff
@@ -68,16 +73,18 @@ func TestConfigFileOtherFields(t *testing.T) {
...
-	}
-
 	if !ctls.equals(&cfg.ClientTLSInfo) {
 		t.Errorf("ClientTLS = %v, want %v", cfg.ClientTLSInfo, ctls)
 	}
 	if !ptls.equals(&cfg.PeerTLSInfo) {
 		t.Errorf("PeerTLS = %v, want %v", cfg.PeerTLSInfo, ptls)
 	}
+
+	testutil.AssertEqual(t, false, cfg.ForceNewCluster, "ForceNewCluster does not match")
```
Comment (lilic) [inline server/embed/config_test.go:83]: @ptabor I noticed we have this test helper function so I used that, let me know if you meant something different? Thank you! (Also adjusted for the forcenewcluster, hope that is okay?)

Comment (ptabor) [reply server/embed/config_test.go]: In the other tests we use "https://github.com/stretchr/testify/assert" that is pretty mature and convenient.

**Review by ptabor — APPROVED** [review]

**Review by hexfusion — APPROVED** [review]
this absolutely makes sense, thanks for fixing it.

File: server/embed/config_test.go (hunk 3 above)
Comment (lilic) [reply server/embed/config_test.go]: Will do that I like that project! I did a quick search so that internal package showed up, is it okay if I open a tech debt issue so we conform to https://github.com/stretchr/testify/assert everywhere?
